### PR TITLE
Blob's snapshots

### PR DESCRIPTION
**Pull request by AlexLamande**
Hello all !

I'm using azure-armrest and I wanted to upgrade it to manage blob snapshots.
Can you please take a look at my PR and tell me if I'm doing it wrong. I'll fix it asap.

Thanks !

### CODE REVIEW

**Comment by AlexLamande** [comment]
Well, I can work on it but as the only revellent thing that is returned is the snap date in headers, I didn't know if returning a ```BlobSnapshot``` made sense.

File: lib/azure/armrest/model/storage_account.rb [hunk 1]
```diff
@@ -249,6 +251,22 @@ def delete_blob(container, blob, key = nil, options = {})
...
+
+        url = File.join(properties.primary_endpoints.blob, container, blob)
+        url += "?comp=snapshot"
+
+        headers = build_headers(url, key, :blob, :verb => 'PUT')
+        headers['Content-Type'] = ''
+        response = RestClient.put(url, '', headers)
+
+        BlobSnapshot.new({
+          :Name => blob,
```
Comment (bzwei) [inline lib/azure/armrest/model/storage_account.rb]: Please align by `=>`
Keys here are strings and lower case (unless the raw API returns capitalized). So they are `'name'` and `'snapshot'`

**Comment by bzwei** [comment]
@AlexLamande please rebase all commits into one.

File: lib/azure/armrest/model/storage_account.rb [hunk 2]
```diff
@@ -249,6 +251,22 @@ def delete_blob(container, blob, key = nil, options = {})
...
+
+        url = File.join(properties.primary_endpoints.blob, container, blob)
+        url += "?comp=snapshot"
+
+        headers = build_headers(url, key, :blob, :verb => 'PUT')
+        headers['Content-Type'] = ''
+        response = RestClient.put(url, '', headers)
+
+        BlobSnapshot.new(
+          :name          => blob,
```
Comment (bzwei) [inline lib/azure/armrest/model/storage_account.rb]: Keys are string like  `'name' => blob`

Comment (AlexLamande) [inline lib/azure/armrest/model/storage_account.rb]: oh ok sorry

**Comment by bzwei** [comment]
LGTM 
@djberg96 please review

**Comment by djberg96** [comment]
Yep, looks good, thanks!

**Comment by bzwei** [comment]
@AlexLamande Thanks.

**Pull request merged**